### ISSUE TITLE
Add alternate forms

### ISSUE THREAD

**Issue by SpiderMath** [DevSnowflake/pokeres#5]
There are alternate forms of a lot of pokemons, which isn't shown here.
I'd like to request adding of those.

Example: Shiny pokemon, Alolan forms of pokemons like exeggutor
![image](https://user-images.githubusercontent.com/71999854/139658325-f52c5fc2-0f31-4bc1-9559-b8355ba5d5bc.png)
Multiple forms of 
![image](https://user-images.githubusercontent.com/71999854/139658442-c6ca5a8b-ff43-4345-9033-f08af8de6cf8.png)

**Comment by twlite** [comment]
Currently the images are mapped by their id so we can't modify existing ones. Maybe creating a new endpoint something like `pokemon.snowflakedev.org/img/alt/:id.png` would be better?

**Comment by SpiderMath** [comment]
Don't think so.
Since as you can see, there maybe multiple forms of the same pokemon like in the case of Wormadam, or even in the Mega forms of Charizard (x and y)
So it would not really fit in the /alt route, I think?

**Comment by twlite** [comment]
we should leave the `/img` route as it is. We can put other forms inside their respective routes. The ones that are currently under `/img` are gen 1 pokemons

**Comment by twlite** [comment]
my bad did not explain it correctly, we do have gen 1 to gen 8 list starting from id `001` to id `898`. Don't see how your other forms would fit in here. And the data we currently have seems to match the **[List of Pokémon](https://en.wikipedia.org/wiki/List_of_Pokémon)**

**Comment by SpiderMath** [comment]
Alright, I get what you mean

Just suggesting, what about we name all the files like `{national_id}-{0 to number_of_forms-1}`
So like, all the base files would be of the name `n-0.png`, and so on, but we could also add the variants as `n-1.png` and stuff, and for megas we could do `n-mega.png` (for Alolan and Hisui variants, we could do `n-alolan.png`/`n-hisui.png`)

After that, we could let the users know about the existence of other forms, like megas and they can add that to their requests
And for the default requests, like `/img/1.png`, we could just return them the `1-0.png`

**Comment by twlite** [comment]
I dont see the problems with using separate route for those **additional forms**

**Comment by bsantosh909** [comment]
Since all pokemon don't have different forms would it be really necessary to have a separate route for it?

Like @SpiderMath suggested it might be better to just add the images as `n-alolan.png` and so on. But Also I don't think renaming the current files as `n-0.png` is good idea.

My suggestion is:

- Keep the existing files (base form image) as it is
- Add the new form images as `-alolan` , `-mega` , `-mega-x` and so on.

Official pokemon site does this in similar way [here](https://assets.pokemon.com/assets/cms2/img/pokedex/full/006.png) and [here](https://assets.pokemon.com/assets/cms2/img/pokedex/full/006_f2.png)

What are your thoughts @DevAndromeda ??

**Comment by twlite** [comment]
@TheLearneer I think your idea is good. Renaming every files is definitely a bad idea. We can use `{id}.mega.png`/`{id}.alolan.png` etc for new images

**Comment by SpiderMath** [comment]
I agree with TheLearneer's idea as well (sounds like a better version of my idea)

**Comment by Shashank3736** [comment]
i have shiny and alolan images. it can be available as 1_s (for shiny type of pokemon id 1) and 1_a (for alolan type of pokemon id 1). BTW not all pokemons have there alolan form but shiny is availble for every pokemon as far as ik
